templates/web/base/auth/general.html

```diff
@@ -16,7 +16,8 @@
 
         <input type="hidden" name="r" value="[% c.req.params.r | html %]">
 
-[% IF NOT oauth_need_email AND (c.config.FACEBOOK_APP_ID OR c.config.TWITTER_KEY) %]
+[% SET social_auth_enabled = (c.config.FACEBOOK_APP_ID OR c.config.TWITTER_KEY) AND NOT c.cobrand.call_hook("social_auth_disabled") %]
```

Comment:
Again, fine, but as it's used three times in three templates, should we factor them as one function (not sure best place, Default.pm and call it as c.cobrand.social_auth_enabled, and have it check the config variables and call call_hook...? Other suggestions welcome).